### ISSUE TITLE
TypeError: Keyword argument not understood 'hidden_size'

### ISSUE THREAD

**Issue by stefan-falk** [tensorflow/models#8381]
I have successfully trained a translation model using a slightly modified implementation of [transformer_main.py](https://github.com/tensorflow/models/blob/master/official/nlp/transformer/transformer_main.py).

I had to run some custom code just in order to adapt the data coming from my `TFRecordDataset`. You can find the *Training Code* further down. The important thing to note here is just, that I am using `tf.keras.callbacks.ModelCheckpoint(save_weights_only=False)` in contrast to transformer_main.py

https://github.com/tensorflow/models/blob/c0c584231b58c14e318bd79b2cf2104ec6e96a25/official/nlp/transformer/transformer_main.py#L424

The `saved_model.pb` gets created but as I am trying to load the model I am getting the `TypeError` exception

> TypeError: Keyword argument not understood 'hidden_size'

This is the code I am using:

```python
import os

import tensorflow as tf
from absl import app
from absl import logging
from official.nlp.transformer import misc


def main():
    model_dir = '/data/asr/models/transformer-translation-v2'
    model = tf.keras.models.load_model(os.path.join(model_dir, 'model-000001'))    
    print('All done.')


if __name__ == '__main__':
    logging.set_verbosity(logging.INFO)
    main()
```

Why isn't this working? `hidden_size` is a parameter coming from the `param_set`. Shouldn't this already be stored in `saved_model.pb`? I cannot explain this `TypeError`.

----

### Training Code

```python
strategy = tf.distribute.MirroredStrategy()

with strategy.scope():

    model = transformer.create_model(params, is_train=True)
    optimizer = tf.keras.optimizers.Adam(0.001)
    model.compile(
        optimizer=optimizer,
        loss='categorical_crossentropy'
    )

    callbacks = [
        tf.keras.callbacks.TensorBoard(log_dir=model_dir),
        tf.keras.callbacks.ModelCheckpoint(
            os.path.join(model_dir, 'model-{epoch:06d}'),  
            monitor='val_loss',
            save_best_only=True,
            save_weights_only=False
        ),
    ]

    model.fit(
        train_dataset,
        epochs=1000,
        steps_per_epoch=10,
        validation_data=dev_dataset,
        validation_steps=10,
        callbacks=callbacks
    )
````

**Comment by saberkun** [comment]
Thank you for the code snippet. Not sure about the answer.  Will ask TF/Keras team.

**Comment by saberkun** [comment]
It would be nice if you can create the issue in tensorflow repo as well.

**Comment by bwang514** [comment]
I wonder is there any update on this bug? I'm having the same problem.

**Comment by gcuder** [comment]
@bwang514 I was able to find a workaround by using the `Seq2SeqTransformer` of this repository and saving it using a `tf.Module`. 

Minimal example
```python
import tensorflow as tf
from official.nlp.modeling.models import seq2seq_transformer

# Define the SaveModule here
class TransformerSaveModule(tf.Module):
    """ Wrapper for saving the Transformer model for inference
    """

    def __init__(self, model):
        super(TransformerSaveModule, self).__init__()
        self.model = model

    @tf.function
    def serve(self, inputs):
        return self.model.call(dict(inputs=inputs))

# Build the model
model = seq2seq_transformer.Seq2SeqTransformer(...)

# Wrap the model into a tf.Module
save_module = TransformerSaveModule(model=model)

# Get the serving signatures
signatures = dict(serving_default=save_module.serving.get_concrete_function(tf.TensorSpec(shape=(None, None), dtype=tf.int32, name="inputs")))

# Save the model
tf.saved_model.save(obj=save_module, export_dir=EXPORT_DIR, signatures=signatures)

# Load the model and test inference
test_input = tf.random.uniform(shape=(4, 256), dtype=tf.int32, maxval=72) # [batch_size, seq_len]

loaded_model = tf.keras.models.load_model(EXPORT_DIR, compile=False)
loaded_model.model.summary()
print(loaded_model.serving(test_input)) 

```

**Comment by saberkun** [comment]
Thank you! @gcuder 

Yes, we have migrated to the new nlp/modeling/. There will be no update to nlp/transformer/ folder. There were bugs and keras problems because the model owner is no longer maintain it. The Seq2SeqTransformer is the new model maintained by us and fully integrated to the training workflow. Sorry about it.
I checked in a unit test to demonstrate the savedmodel inference workflow: https://github.com/tensorflow/models/blob/master/official/nlp/modeling/models/seq2seq_transformer_test.py#L104

If @bwang514 you see the new model works, we will close this issue.
